crates/core_simd/src/vector.rs

```diff
@@ -44,6 +44,39 @@ use crate::simd::{LaneCount, Mask, MaskElement, SupportedLaneCount};
 ///
 /// [`Wrapping<T>`]: core::num::Wrapping
 ///
+/// # Layout
+/// `Simd<T, N>` has a layout similar to `[T; N]` (identical "shapes"), but with a higher alignment.
+/// `[T; N]` is aligned to `T`, but `Simd<T, N>` will have an alignment based on both `T` and `N`.
+/// It is thus sound to transmute from `Simd<T, N>` into `[T; N]` but not vice versa.
+///
+/// # ABI Bugs
+/// Due to an unfortunate bug in the Rust ABI, `Simd<T, N>` does not participate in argument-passing in SIMD registers,
```

Comment:
I wouldn't consider this a bug so much as the current definition of the ABI... It's certainly intentional behavior, even if unfortunate.

Reply:
A feature is a bug we don't intend to correct,
a bug is a feature we intend to remove. :^)

Reply:
I still think it's misleading.  It's definitely not a mistake or unintended behavior, it's just a limitation.

Reply:
It's "unintended" in the sense that it's a non-zero cost fix for a compiler smartness limitation.

Reply:
Bug carries the connotation that your programs will not work, which is the opposite of what this does, it's specifically a solution for avoiding mismatched calling conventions, which _is_ a bug and will cause your programs to not work.

Reply:
perhaps "problem with the Rust ABI"?

Reply:
I would just note that it's not optimal, but it's avoiding a catastrophic bug.

Reply:
(i mean sorta, but doesn't it happen even when the calling conventions are known to match?)

Reply:
On architectures where they're known to match (like wasm) I don't believe they're passed by reference.  But on an architecture with multiple possible conventions, the callee can't possibly know all of it's caller's conventions.

Reply:
```suggestion
/// # ABI "Features"
/// Due to Rust's safety guarantees, `Simd<T, N>` is passed to and from functions via memory, not SIMD registers, 
```

Reply:
maybe phrase it "Due to Rust's safety guarantees, `Simd<T, N>` is *currently* passed to and from functions via memory, not SIMD registers,"

Reply:
And that's incorrect, Caleb: it happens on wasm too.